lib/Log/Any/Adapter/DERIV.pm

```diff
@@ -656,6 +659,55 @@ sub clear_context {
     $adapter_context= undef;
 }
 
+=head2 mask_sensitive
+
+Mask sensitive data in the message and logs error in case of failure
+
+=over 4
+
+=item * C<$message> string - The message to be masked
+
+=back
+
+Returns string - The masked message
+
+=cut
+
+sub mask_sensitive {
+    my ($message) = @_;
+
+    # Define a lookup hash reference for all sensitive data regex patterns to be logged
+    my $sensitive_patterns = {
+        'PII' => [
+            qr/\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b/,  # Email
+        ],
+        'Sensitive' => [
+            qr/\b(?:token|key|oauth[ _-]?token)\s*[:=]\s*([^\s]+)/i, # token or api key
```

Comment:
Wouldn't it be covered by the next category?

Reply:
this is being covered in next category as well but here we are checking : = as boundaries as well with these words, the next is covering the actual token value (how it will look like)

Reply:
Suppose I've a string like `ouath token is a1-abcd1234`. it should be `oauth token is ******`. sorry didn't get what we are trying to achieve, can you provide an example

Reply:
IIUC, it's to mask unknown patterns?
Regarding the pattern itself, I think `oauth[ _-]token` will be covered by `token`
you can also try the `look behind` pattern `(?<=[key |token])` this way the `s///` will not replace it.

Reply:
@hani-deriv you mean like this 
`s/(?<=[key|token])\b(?:token|key|oauth[ _-]?token)\s*[:=]\s*([^\s]+)/'*' x length($&)/ige;`

Reply:
```
qr/\b(?<=[key |token|])\s*[:=]\s*([^\s]+)/i
```

Reply:
I tried this yesterday @hani-deriv but the result is partially redacting the token value for example 
if the message is  "The token = 81983nncjashdHAH" it showing something like "The token = *************HAH" its just an example but I cant help to mimic this is test so I have kept it as is the previous one